unblob/handlers/archive/instar/hd.py

```diff
@@ -0,0 +1,105 @@
+import io
+import struct
+from pathlib import Path
+from typing import Optional
+
+from structlog import get_logger
+
+from unblob.file_utils import InvalidInputFormat, iterate_patterns
+from unblob.models import Extractor, File, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+
+EOCD_RECORD_HEADER = 0x9054B50
+
+
+class InstarHDExtractor(Extractor):
+    def extract(self, inpath: Path, outpath: Path):
+        with open(inpath, "rb") as file:
+            outfile = outpath.joinpath(f"{inpath.name}")
+            with open(outfile, "wb") as out_file:
+                headers = file.read()
+
+                if b"\x50\x4b\x03\x07" in headers:
+                    headers = headers.replace(b"\x50\x4b\x03\x07", b"\x50\x4b\x03\x04")
+
+                if b"\x50\x4b\x05\x09" in headers:
+                    headers = headers.replace(b"\x50\x4b\x05\x09", b"\x50\x4b\x05\x06")
+
+                if b"\x50\x4b\x01\x08" in headers:
+                    headers = headers.replace(b"\x50\x4b\x01\x08", b"\x50\x4b\x01\x02")
```

Comment:
I would move this into a dedicated function named `fix_headers_magic`. I would also remove the `if` condition (the `if` will traverse the bytearray a first time, and then if something is found it will be traversed a second time).

I would also assign the byte strings to constants so that the reader understands what you are doing. Something like:

```python
CD_FILE_HEADER_ZIP = b"\x50\x4b\x03\x04"
CD_FILE_HEADER_INSTAR = b"\x50\x4b\x03\x07"
```